### DISCUSSION TITLE
qustion about INTERSECT in SqlBase.g4

### DISCUSSION THREAD

**Discussion by boson234** [trinodb/trino#14098] [General]
why not merge these two into one in SqlBase.g4?
![image](https://user-images.githubusercontent.com/34983864/189566562-245fae10-7845-4ba9-b4bd-9673e799b2e5.png)

just like 
` left=queryTerm operator=(INTERSECT | UNION | EXCEPT) setQuantifier? right=queryTerm  #setOperation`

**Comment by martint** [comment]
INTERSECT has different precedence than EXCEPT and UNION. In Antlr, when there are left-recursive rules, the order of the alternatives determines the precedence.
